docs/_sandboxenv-interface.md

```diff
@@ -18,6 +18,8 @@ class SandboxEnvironment:
             decoding the command output.
           PermissionError: If the user does not have
             permission to execute the command.
+          SandboxOutputLimitExceededError: If an output
```

Comment:
Could both of these be `OutputLimitExceededError`? (for consistency w/ each other and to keep them general so they could in theory be used elsewhere)

Reply:
Yep, very happy to simplify this! I agree with all your comments - thank you.

A question this raises: Should we keep the `OutputLimitExceededError.truncated_result` field? If so, it will have to become `truncated_result: ExecResult[str] | None` (because `read_file()` doesn't deal in `ExecResult`). Or make `OutputLimitExceededError` a `Generic[T]` class which has a `truncated_result` field of type `T`.

I think Ed raised a valid point about the truncated output possibly being useful to tool developers to show to the model, but if it adds significant complexity or ties this exception too closely to one particular use case (`SandboxEnvironment.exec()`) I'd be open to removing it.

Reply:
Can the truncated result just be reduced to a `str` in both cases?

Reply:
It certainly could, but it doesn't feel like it should be the `SandboxEnvironment`'s responsibility to decide how to format an `ExecResult[str]` into a `str`.

For example, some tools format stdout/stderr as json. Some just concat them.

If we go for the `truncated_output: str` approach, I think we'd just want to concat stdout and stderr; acknowledging that this isn't how all tool authors would want it presented.

Reply:
I am just thinking for the sake of simplification and uniformity, just concatenating them it seems to me would be fine. If people want/need more they will ask for it and we can consider it in context of their requirements.

Reply:
Good call! Changed as suggested.

Note: I have not made the `read_file()` include the 100 MiB truncated output in the exception. I'm worried about:
a) 100 MiB strings unnecessarily bloating inspect's memory footprint
b) the massive field on this exception inadvertently making its way into logs (either py `logging` or the .eval logs)
c) any performance implications of exceptions rooting large amounts of memory
d) it won't allow us to "fail early" if the agent tries to ready a 100+ MiB file.

Reply:
Agree w/ this 100%